distribution/src/bin/elasticsearch.bat

```diff
@@ -68,6 +81,21 @@ IF "%checkpassword%"=="Y" (
   )
 )
 
+IF "!enrolltocluster!"=="Y" (
+    CALL "%~dp0elasticsearch-env.bat" || exit /b 1
+    CALL "%~dp0x-pack-env"
```

Comment:
This is going to pollute the classpath we use to start Elasticsearch later. Why can't we run `elasticsearch-enroll-node.bat`?

Reply:
Because of what I describe right above in https://github.com/elastic/elasticsearch/pull/77718/files#r710175659. If you have any suggestions or if 'm missing something I'm more than eager to know , I had already asked you for advice :)

Reply:
I think we need to eitiher stash/restore the classpath/etc env vars, or call the cli so this all happens in a separate process. I prefer the latter, as it is not as error prone with possibly missing an env var that needs to be restored.

Reply:
Yes, I know that this is the alternative but I was wondering if you had any ideas around what I mention in that comment, which are the issues I stumble upon when calling the CLI :

> - This CALL here triggers a "multiple garbage collectors defined" error message
> - I know of no way to pass the keystore password as stdin to this execution. We can call the CLI batch script directly instead of this invocation if it helps ( I couldn't figure out how )

Reply:
I'll look into it again fwiw, but just wondering if anything stood out

Reply:
> We can call the CLI batch script directly 

This is exactly what I'm suggesting. I don't know the best approach for doing that call out, hoping that @williamrandolph can provide some suggestions.

Reply:
Understood. That was probably not communicated clearly from my side but this was my original approach and it came with the two drawbacks that I mention above and that's way I changed it to the current approach. 

I'll wait for William to chime in

Reply:
If this ends up being too problematic, we are thinking about ditching the support for pre-encrypted keystores whcih is the main pain point here ( if we don't need to pass the keystore password in stdin then we can simply `CALL` the batch script ) 

The use case where this would be useful is where a user would want to create and password protect their keystore _before_ the initial start of the node but leave all other security configuration on us. We think this is a corner case because:

- When a user knows that they want to encrypt the keystore, and at that even before the node starts for the first time, they are probably not the user persona that is the main target group of the security on by default project
- Users can always password protect the keystore _after_ auto-configuration happens ( which does require a restart of the node though )